crates/client/rpc/src/lib.rs

```diff
@@ -943,48 +944,88 @@ where
             }
         }
 
+        fn extract_fee_and_contract_address(
+            events: &Vec<starknet_core::types::Event>,
+            sequencer_address: &Felt252Wrapper,
+        ) -> (FieldElement, Option<FieldElement>) {
+            let transfer_event_key = starknet_core::utils::get_selector_from_name("Transfer").unwrap();
+            let find_transfer_fee_event = events.iter().find_map(|event| {
+                event.keys.iter().find(|&&key| key == transfer_event_key)?;
+                event.data.get(1).and_then(|send_to| (*send_to == sequencer_address.0).then_some(()))?;
+                Some(event)
+            });
+
+            match find_transfer_fee_event {
+                Some(transfer_fee_event) => {
+                    let actual_fee = Felt252Wrapper(transfer_fee_event.data[2].clone()).0; // ony low bits, see execute_fee_transfer() why
```

Comment:
Can you explain why here?

Reply:
you mean why only low bits? I copied comment from `execute_fee_transfer()` here